library/src/main/scala/scala/scalajs/js/Union.scala

```diff
@@ -66,6 +79,14 @@ object | { // scalastyle:ignore
   implicit def from[A, B1, B2](a: A)(implicit ev: Evidence[A, B1 | B2]): B1 | B2 =
     a.asInstanceOf[B1 | B2]
 
+  /** Upcast `F[A]` to `F[B]`.
+   *
+   *  This needs evidence that `F[A] <: F[B]`.
+   */
+  implicit def fromTypeConstructor[F[_], A, B](a: F[A])(
+      implicit ev: Evidence[F[A], F[B]]): F[B] =
```

Comment:
Yep. In fact, ideally, there would be only one entry point conversion
```scala
implicit def from[A, B](a: A)(implicit ev: Evidence[A, B]): B
```
Both the actual `from` and `fromTypeConstructor` are workaround for the "diverging implicit" limitations of scalac, which happen worth the ideal universal `from`.